envs/numpy_env.py

```diff
@@ -10,39 +11,56 @@ def __init__(self, seed, workspace, max_steps=10, heightmap_size=250, render=Fal
 
     self.render = render
     self.offset = self.heightmap_size/20
+    self.valid = True
```

Comment:
What is this valid flag used for?

Reply:
It's for stacking. If one block is placed overlapping another, and it is not a valid stacking, it will be set True

Reply:
I assume you meant it would be set to False for a non-valid stacking?